### ISSUE TITLE
release v0.2.4

### ISSUE THREAD

**Issue by joshrabinowitz** [sobolevn/git-secret#143]
v0.2.4 has been merged but needs to be released. What needs to happen for the release?

Happy to help

**Comment by sobolevn** [comment]
I guess we should fix CI before releasing the new version.
Right now it uses so many tools! `Docker`, `ansible`, `kitchen`, so it is almost impossible to maintain them together.

I think dropping everything right now is the only way to go.
So, we need to specify envs and `gnupg` versions we want to test.

I think, that we can test only two envs for now: `fedora` and `debian`.
And two `gnupg` versions: `1` and `2`. What gives us just 4 different setups.

Later we can think of better testing strategies.
@hurricanehrndz what do you think about it?

**Comment by notjames** [comment]
Did you fix the VERSION bug? Just curious.

**Comment by joshrabinowitz** [comment]
I fixed it moving forward, but not retroactively for 0.2.3.

**Comment by joshrabinowitz** [comment]
See also: #120

**Comment by simbo1905** [comment]
Our team has spent a few hours with two macos and two ubuntu users working out compatibility issues. The macos version installed by homebrew says it's v0.2.2 when it is v0.2.3 and inserts colons that the ubuntu release does not recognize. We almost gave up. Releasing v0.2.4 would hopefully fix git secret for heterogeneous teams. Any chance of that happening soon? We can help test.

**Comment by joshrabinowitz** [comment]
@simbo1905 thanks for your note. 
The big blocker on getting 0.2.4 out are the issues in #120 . 
We could use some help in getting a new release out.

**Comment by simbo1905** [comment]
Fixing the other things uncovered #189

**Comment by simbo1905** [comment]
all fixed on my fork https://travis-ci.org/simbo1905/git-secret/builds/389940226

**Comment by simbo1905** [comment]
The deb package deployed! 🎉

https://bintray.com/sobolevn/deb/git-secret

The rpm didn't

https://travis-ci.org/sobolevn/git-secret

**Comment by simbo1905** [comment]
What are the steps to deploy to brew for MacOS?

**Comment by simbo1905** [comment]
RPM is deployed https://bintray.com/sobolevn/rpm/git-secret 

So do we now need a brew release to close this?

**Comment by joshrabinowitz** [comment]
@simbo1905 

The page http://git-secret.io/installation mentions homebrew, deb, and rpm, so I'd say we do need a brew release to close this.

Do we want to update the CHANGELOG.md file more?

Also probably we should consider making the 'github release' a prereq to closing this issue.

@sobolevn anything else you want to see done as part of this issue?

**Comment by sobolevn** [comment]
Yeap, `brew` is a manual process. But we can submit a patch to `homebrew` repo after the package is released.

Yes, github release and a few notes about the latest fixes to the `CHANGELOG.md` would be great.

**Comment by simbo1905** [comment]
@sobolevn When you do the manual `brew` release for 0.2.4 add the detailed commands to `CONTRIBUTING.md` and I will try to find some time to automate the process.

**Comment by simbo1905** [comment]
This ruby project generates a change log https://github.com/github-changelog-generator/github-changelog-generator perhaps we should be running that on the CI master?

**Comment by joshrabinowitz** [comment]
I'd say for this release we should continue maintaining CHANGELOG.md manually.

**Comment by simbo1905** [comment]
@sobolevn grepping the codebase I don't see any solid hits about how `brew` is done manually to install it. `make install` bombs on my macos. can you please give me some hints as to what you to do make a brew package?

**Comment by sobolevn** [comment]
@simbo1905 this code is located in the `brew` repo: https://github.com/Homebrew/homebrew-core/blob/master/Formula/git-secret.rb

```ruby
 def install
    system "make", "build"
    system "bash", "utils/install.sh", prefix
  end
```

**Comment by simbo1905** [comment]
@sobolevn 

> $ make clean
> $ make build
> $ utils/install.sh /usr/local/bin
> cp: /Users/Shared/git-secret/utils/bin/*: No such file or directory
> $ ls -al /Users/Shared/git-secret/utils/bin/
> total 0

**Comment by joshrabinowitz** [comment]
@simbo1905 @sobolevn I think we should call 0.2.4 done and make a release. 

I've set up a 'project' https://github.com/sobolevn/git-secret/projects/1 (release 0.2.4) for this purpose. Feel free to add/remove tasks from the project.

**Comment by joshrabinowitz** [comment]
@simbo1905 can you tag and release 0.2.5 (with all the needed changes for .deb, .rpms, brew packages, etc?) 
Let me know if you want me to update CHANGELOG.md and/or bump version numbers.

**Comment by simbo1905** [comment]
🎉🎉🎉🎉🎉🎉🎉 

@ilovezfs has merged our brew PR! https://github.com/Homebrew/homebrew-core/commit/13e2ea914c0fbbc1a96df41bbf5bfabd7b35289b

@sobolevn we simply need you to update the latest release on github to be v0.2.4 to close this issue.

**Comment by sobolevn** [comment]
@simbo1905 which commit should be the latest to be included in the release?

**Comment by simbo1905** [comment]
This one https://github.com/Homebrew/homebrew-core/blob/13e2ea914c0fbbc1a96df41bbf5bfabd7b35289b/Formula/git-secret.rb#L13 

If we add it to the release that brew is now patching (as per that line there) would it be idempotent and not break brew that "live" brew formula? Let's ask our friendly brew maintainer! 

hey @ilovezfs can you help us out on this brew question. I put that patch into the formula. The formula references a release. @sobolevn is thinking to cheery pick it into the release and update the url that brew is patching. Is that safe?

**Comment by ilovezfs** [comment]
@simbo1905 that's no safe, no, as it will change the checksum of https://github.com/sobolevn/git-secret/archive/v0.2.4.tar.gz. What would be safe is releasing it as 0.2.4.1 or 0.2.5

**Comment by joshrabinowitz** [comment]
Do we consider 0.2.4 'released'? 
Since it's not safe to try to fix 0.2.4, should we go for 0.2.5?

**Comment by simbo1905** [comment]
0.2.4 is released. I feel any actual code fixes should go out as 0.2.5. Any trivial doc or installation changes could go out as 0.2.4.1. I will close this issue and let's raise a new one for whatever comes next.
